### ISSUE TITLE
IDE0004 incorrectly suggests int to enum cast in switch expression is redundant

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn#37953]
**Version Used**: 
3.3.0-beta3-19406-05+a1905991543bed104f7f7f0842aca2b65d263b87

**Steps to Reproduce**:
```C#
using System;

class Program
{
    public static void Main() { }

    public static string GetValue(int value) =>
        (DayOfWeek)value switch
        {
            DayOfWeek.Monday => "Monday",
            _ => "Other",
        };
}
```

**Expected Behavior**:
IDE0004 doesn't trigger.

**Actual Behavior**:
IDE0004 triggers and suggests removing the `(DayOfWeek)` cast:
![image](https://user-images.githubusercontent.com/2642209/62953971-8850aa80-bdbc-11e9-91ec-4579b93ec02c.png)
Removing the cast then produces a compiler error:
![image](https://user-images.githubusercontent.com/2642209/62953997-9acae400-bdbc-11e9-9076-bd908dc86965.png)
